### DISCUSSION TITLE
Field "Options Editor" - not possible to set empty "site"

### DISCUSSION THREAD

**Discussion by rob-c-baker** [solspace/craft-freeform#281] [Feature Requests]
When setting a field of (internal) type `CheckboxGroupField` and setting it's options to be dynamic based on a `source` of Categories, within the "Options Editor" it is not possible to set the `site` field to `null` to allow the code in `Solspace\Freeform\Services\FieldsService::getOptionsFromSource()` around line 375 to pick up the `siteId` from Craft's current site.

`FieldsService::getOptionsFromSource()` has:

```php
$siteId = $config->getSiteId() ?? \Craft::$app->sites->currentSite->id;
```

Which should ensure the database query criteria picks up the current site on the front end when getting options for the field but the field / form CMS interface only allows picking specific sites, not allowing it to get the "current" one.

This is causing an issue for us as we have different sets of categories enabled /disabled on different sites and the form should be reflecting this in the options available to this checkbox field.

**Steps to reproduce**

1. Have a multi-site Craft set-up
2. Create a category group and some contained categories
3. Some categories should be enabled and some disabled in at least 2 different sites
4. Create a (Freeform) form with a "checkbox group" field
5. Place the field on a form and set the field to get it's "options" from a source of "Categories" (probably any element based selection would have the same effect here)
6. View the form on the front-end - the list of categories / checkboxes will only be from one of the sites

**Expected behaviour**

At step 5 above, it should be possible to **not** set a specific **site** so that the options are loaded dynamically based on the categories enabled within whatever site the end-user is using.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 3.7.13
- Freeform Version: 3.12.12.1
- Freeform Edition: Pro
- Fresh Install or Upgrade: Upgrade

**Comment by kjmartens** [comment]
Thanks for your feedback on this. I've made a note about it and we'll see if this is a change we can easily make in the near future. Until then, the other option you can do is set this field to a regular Text field type to accept entry element ID's/handles/titles that are generated inside your template. 🙂
